modules/node_modules/@ncigdc/components/CartDownloadDropdown.js

```diff
@@ -0,0 +1,161 @@
+// @flow
+
+import React from 'react';
+import { compose, withState } from 'recompose';
+import { connect } from 'react-redux';
+
+import DownloadButton from '@ncigdc/components/DownloadButton';
+import LoginButton from '@ncigdc/components/LoginButton';
+
+import DownloadIcon from 'react-icons/lib/fa/download';
+import DownCaretIcon from 'react-icons/lib/fa/caret-down';
+
+import { setModal } from '@ncigdc/dux/modal';
+
+import Dropdown from '@ncigdc/uikit/Dropdown';
+import Button from '@ncigdc/uikit/Button';
+import { Column, Row } from '@ncigdc/uikit/Flex';
+
+import { withTheme } from '@ncigdc/theme';
+
+import download from '@ncigdc/utils/download';
+/*----------------------------------------------------------------------------*/
+
+const styles = {
+  button: theme => ({
+    backgroundColor: 'transparent',
+    color: theme.greyScale2,
+    justifyContent: 'flex-start',
+    borderRadius: '0px',
+    marginLeft: '0px',
+    ':hover': {
+      backgroundColor: theme.greyScale6,
+    },
+  }),
+  iconSpacing: {
+    marginRight: '0.6rem',
+  },
+};
+
+const downloadCart = (files, dispatch, setActive) => {
+  const controlledFiles = files.filter((file) => file.access === 'controlled');
+
+  if (controlledFiles.length) {
+    dispatch(setModal(
+      <Column
+        style={{ margin: '-15px' }}
+      >
+        <div className="modal-header">
+          <h4> Access Error </h4>
+        </div>
+        <div className="modal-body">
+          <p>You are attempting to download files that you are not authorized to access.</p>
+          <p>
+            <span className='label label-success'>
+              {files.length - controlledFiles.length}
+            </span> files that you are authorized to download.
+          </p>
+          <p>
+            <span className='label label-danger'>
+              {controlledFiles.length}
+            </span> files that you are not authorized to download.
+          </p>
+          <p>
+            Please <LoginButton />
+          </p>
+        </div>
+        <div
+          className="modal-footer"
+          style={{
+            display: 'flex',
+            flexDirection: 'row',
+            justifyContent: 'flex-end',
+          }}
+        >
+          <Button
+            onClick={() => dispatch(setModal(null))}
+            style={{ margin: '0 10px' }}
+          >
+            <span>Cancel</span>
+          </Button>
+          <Button
+            onClick={() => downloadCart(files.filter((file) => file.access === 'open'), dispatch)}
+            style={{ margin: '0 10px' }}
+          >
+            <span>Download {files.length - controlledFiles.length} authorized files</span>
+          </Button>
+        </div>
+      </Column>
+    ));
+  } else {
+    dispatch(setModal(null));
+
+    setActive(true);
```

Comment:
Clicking on Download -> Cart
![image](https://cloud.githubusercontent.com/assets/743976/23185559/cc6debca-f851-11e6-914a-f5fa01fd8958.png)
Results in 
```
CartDownloadDropdown.js:93 Uncaught TypeError: setActive is not a function
    at downloadCart (http://local.dcc.icgc.org:8080/js/bundle.js:242131:5)
```


I think this function should maybe be in a handler so you don't have to keep track of all the incidental props you need to pass in the places where it's used

https://github.com/acdlite/recompose/blob/master/docs/API.md#withhandlers